### PR TITLE
Alerts for 2017-04-11.a security issue (fixed in 1.0.8-1)

### PR DESCRIPTION
**Pull request by daira**
Alert 1002 (versions 1.0.0-1.0.2 inclusive).
Alert 1003 (versions 1.0.3-1.0.8 inclusive).

Signed-off-by: Daira Hopwood <daira@jacaranda.org>

### CODE REVIEW

**Comment by nathan-at-least** [comment]
Code review ACK on the versions, priorities, and text. I guess there's no support for non-English-speaking users?

**Comment by daira** [comment]
> I guess there's no support for non-English-speaking users?

There is not :-(

File: src/sendalert.cpp [hunk 1]
```diff
@@ -71,29 +71,35 @@ void ThreadSendAlert()
...
     alert.strComment    = "";
-    alert.strStatusBar  = "Your client is out of date and potentially vulnerable to denial of service. Please update to the most recent version of Zcash (1.0.5). More info at: https://z.cash/support/security.html";
-    alert.strRPCError   = "Your client is out of date and potentially vulnerable to denial of service. Please update to the most recent version of Zcash (1.0.5). More info at: https://z.cash/support/security.html";
+    alert.strStatusBar  = "Your client is out of date and potentially vulnerable to blockchain disruption. Please update to the most recent version of Zcash and use -reindex to ensure integrity of your blockchain state. More info at: https://z.cash/support/security/";
+    alert.strRPCError   = alert.strStatusBar;
 
     // Set specific client version/versions here. If setSubVer is empty, no filtering on subver is done:
     // alert.setSubVer.insert(std::string("/MagicBean:0.7.2/"));
-	alert.setSubVer.insert(std::string("/MagicBean:1.0.3/"));
+    const std::vector<std::string> useragents = {"MagicBean", "BeanStalk", "AppleSeed", "EleosZcash"};
```
Comment (str4d) [inline src/sendalert.cpp]: Do we know that all of these are using the affected codebase? It's a fair assumption, but we should probably check.

Comment (daira) [inline src/sendalert.cpp]: These were the user agents I saw at https://explorer.zcha.in/network that appeared to be using the same version numbers as the official source.

**Comment by nathan-at-least** [comment]
@zkbot r+

**Pull request merged**